ph5/core/ph5api.py

```diff
@@ -1419,9 +1438,61 @@ def get_extent(self, das, start=None, end=None, component=None):
         latest_epoch = (latest_epoch_start +
                         (float(new_das_t[-1]['sample_count_i'])
                          / true_sample_rate))
-
+        self.forget_das_t(das)
         return earliest_epoch, latest_epoch
 
+    def get_availability(self, das,
+                         sample_rate, component,
+                         start=None, end=None):
+        '''
+        Required: das, sample_rate and component
+        Optional: Start time, End time
+        :param das: das serial number
+        :param start: start time epoch
+        :param end:  end time epoch
+        :param sample_rate: sample rate
+        :param component: component channel number
+        :return: list of tuples (sample_rate, start, end)
+        '''
+        if not component:
+            raise ValueError("Comonent required for get_avilability")
+        if not sample_rate:
```

Comment:
I've been considering this for other reasons as well. The das_t only has a start time and number of samples in the window. It would make things way easier if it also had and end time available. Then there would be no need to do these calculations unless it's an old experiment and the end time isn't there.

For now until we get approval from standing committee to add new fields to PH5 we can do the same thing and have 0 sample rate channel/component_num set as a negative number. If it is negative then the sample_count is the length of the window in microseconds instead of number of samples  and the latest_epoch can still be calculated without a divide by zero error

Reply:
Actually for now i'm totally over thinking this. ALL the miniSEED logs I have found are just timestamped with each entry and length of 0 so every window has no length just a start time. So until approval for new end time in das_t, then if sample_rate is 0 end is the same as start so no need to do the division in that case just a check if sample rate is 0 or not